src/decays/experimental_constraints.cpp

```diff
@@ -0,0 +1,579 @@
+      // otherwise it returns GSL_EINVAL status.
+      // In this case we randomly try different m from [max(a, 0.01m), min(b, 100m)]
+      // until status != GSL_EINVAL
+      gsl_error_handler_t * _error_handler = gsl_set_error_handler_off();
+      status = gsl_min_fminimizer_set (sGSL, &F, m, a, b);
+      if (status == GSL_EINVAL) {
+         std::random_device rd;
+         std::mt19937 gen(rd());
+         std::uniform_real_distribution<> dis(std::max(a,1e-2*m), std::min(b,1e+2*m));
+         int iterCount = 0;
+         do {
+            m = dis(gen);
+            status = gsl_min_fminimizer_set (sGSL, &F, m, a, b);
+            iterCount++;
+         } while (status == GSL_EINVAL && iterCount < 100);
+      }
+      gsl_set_error_handler (_error_handler);
+
+      static constexpr double mass_precision = 1e-4;
+
+      do
+      {
+           iter++;
+           status = gsl_min_fminimizer_iterate (sGSL);
+
+           m = gsl_min_fminimizer_x_minimum (sGSL);
+           a = gsl_min_fminimizer_x_lower (sGSL);
+           b = gsl_min_fminimizer_x_upper (sGSL);
+      }
+      while (std::abs(1. - sm.get_physical().Mhh/mass) > mass_precision && iter < max_iter);
+
+      gsl_min_fminimizer_free (sGSL);
+
+      if (const double diff = std::abs(1. - sm.get_physical().Mhh/mass); diff > mass_precision) {
+         throw std::runtime_error("Normalized Higgs effective couplings: Cannot find a SM equivalent of " + name +
+                     " after " + std::to_string(iter+1) + "/" + std::to_string(max_iter) + " iterations. "
+                     "Mass difference: " + std::to_string(mass) + " GeV (BSM) vs " +
+                     std::to_string(sm.get_physical().Mhh) + " GeV (SM) for λSM = " + std::to_string(m) +
+                     ". Difference: " + std::to_string(100*diff) + "%. ");
+      }
+
+
+}
+} // anonymous
+
+double chi2_to_pval(double chi2BSM, double chi2SM) {
+   boost::math::chi_squared dist(2);
+   const double pval = chi2BSM<=chi2SM ? 1 : boost::math::cdf(complement(dist, chi2BSM-chi2SM));
+   return pval;
+}
+
+EffectiveCoupling_list get_normalized_effective_couplings(
+   EffectiveCoupling_list const& bsm_input,
+   Physical_input const& physical_input,
+   softsusy::QedQcd const& qedqcd,
+   Spectrum_generator_settings const& spectrum_generator_settings,
+   FlexibleDecay_settings const& flexibledecay_settings)
+{
+   // make sure we don't compute input for the EFFHIGGSCOUPLINGS block in the
+   // built in SM
+   auto flexibledecay_settings_ = flexibledecay_settings;
+
+   EffectiveCoupling_list _bsm_input;
+   for (auto const& el : bsm_input) {
+
+      const double mass = el.mass;
+      // in the SM, λ = (mh/v)^2/2
+      // for mh > 700 GeV this gives λ > 4
+      // it probably makes no sense to use coupling strengh modifiers in this case so we skip those particles
+      // On the other hand there's a problem with finding a SM equivalent of very light states
+      if (mass > 650 || mass < 1) continue;
```

Comment:
I'd recommend to also print a warning here. We could use `WARNING` or `VERBOSE_MSG`.

Reply:
Done in bae5f3cea1ae2364a821757bf2a81c158a9f60e3. I decided to go with `VERBOSE_MSG` because it will happen in almost all models so it doesn't make sense to pollute the output. I've added the following message
```
Cannot construct normalization for effective coupling of hh(2) with mass of 19856.713128 GeV
```

Reply:
Thanks!